pyfive/h5d.py

```diff
@@ -218,18 +247,33 @@ def convert_slice(aslice):
             else:
                 yield convert_selection(out_selection)
 
-   
 
     ##### The following property is made available to support ActiveStorage
     ##### and to help those who may want to generate kerchunk indices and
     ##### bypass the iterator methods.
     @property
     def index(self):
         """ Direct access to the chunk index, if there is one. This is a ``pyfive`` API extension. """
-        if self._index is None:
-            raise ValueError('No chunk index available for HDF layout class {self.layout}')
-        else:
-            return self._index
+        # can't use init_chunk_check because that would be an infinite regression
+        if self.layout_class != 2:
+            raise TypeError("Data is not chunked") 
+        if not self._index:
+            self._build_index()
+        return self._index
+        
+
+    ##### This property is made available to help understand object store performance
+    @property
+    def btree_range(self):
```

Comment:
Could we also have `def first_chunk(self):` and `def n_chunks(self):` properties?